Floquet using v5's Solver interface (#2034)
qutip/qutip: doc/guide/dynamics/dynamics-floquet.rst

   >>> args = {'w': omega}
   >>> H = [H0, [H1, 'sin(w * t)']]

> The :math:`t=0` Floquet modes corresponding to the Hamiltonian :eq:`eq_driven_qubit` can then be calculated using the :class:`qutip.FloquetBasis` class, which encapsulated the Floquet modes and the quasienergies

Reviewer: ```suggestion
The :math:`t=0` Floquet modes corresponding to the Hamiltonian :eq:`eq_driven_qubit` can then be calculated using the :class:`qutip.FloquetBasis` class, which encapsulates the Floquet modes and the quasienergies:
```

   [[-0.37793106-0.00431336j]
    [-0.89630512+0.23191946j]]]

The purpose of calculating the Floquet modes is to find the wavefunction solution to the original problem :eq:`eq_driven_qubit` given some initial state :math:`\left|\psi_0\right>`.
To do that, we first need to decompose the initial state in the Floquet states, using the function :meth:`FloquetBasis.to_floquet_basis`

.. plot::
   :context:

   >>> psi0 = rand_ket(2)
   >>> f_coeff = floquet_basis.to_floquet_basis(psi0)
   >>> f_coeff # doctest: +SKIP
   [(-0.645265993068382+0.7304552549315746j),
   (0.15517002114250228-0.1612116102238258j)]

> and given this decomposition of the initial state in the Floquet states we can easily evaluate the wavefunction that is the solution to :eq:`eq_driven_qubit` at an arbitrary time :math:`t` using the function :meth:`FloquetBasis.from_floquet_basis`

Reviewer: ```suggestion
and given this decomposition of the initial state in the Floquet states we can easily evaluate the wavefunction that is the solution to :eq:`eq_driven_qubit` at an arbitrary time :math:`t` using the function :meth:`FloquetBasis.from_floquet_basis`:
```

Pre-computing the Floquet modes for one period
----------------------------------------------

> When evaluating the Floquet states or the wavefunction at many points in time it is useful to pre-compute the Floquet modes for the first period of the driving with the required resolution.
> In QuTiP the function :func:`qutip.floquet.floquet_modes_table` calculates a table of Floquet modes which later can be used together with the function :func:`qutip.floquet.floquet_modes_t_lookup` to efficiently lookup the Floquet mode at an arbitrary time.

Reviewer: ```suggestion
When evaluating the Floquet states or the wavefunction at many points in time it is useful to pre-compute the Floquet modes for the first period of the driving with the required times.
The list of times to pre-compute modes for may be passed to :class:`FloquetBasis` using `precompute=tlist`, and then `:meth:`FloquetBasis.from_floquet_basis` and :meth:`FloquetBasis.to_floquet_basis` can be used to efficiently retrieve the wave function at the pre-computed times.
```

> The following example illustrates how the example from the previous section can be solved more efficiently using these functions for pre-computing the Floquet modes.

Reviewer: ```suggestion
The following example illustrates how the example from the previous section can be solved more efficiently using these functions for pre-computing the Floquet modes:
```

    gamma1 = 0.1
    def noise_spectrum(omega):
        return (omega>0) * 0.5 * gamma1 * omega/(2*pi)

The other parameters are similar to the :func:`qutip.mesolve` and :func:`qutip.mcsolve`, and the same format for the return value is used :class:`qutip.solve.solver.Result`. The following example extends the example studied above, and uses :func:`qutip.floquet.fmmesolve` to introduce dissipation into the calculation

.. plot:: guide/scripts/floquet_ex3.py
   :width: 4.0in
   :include-source:

> Finally, :func:`qutip.solver.floquet.fmmesolve` consider the ``e_ops`` to always be in the laboratory basis, thus we can simply use:

Reviewer: ```suggestion
Finally, :func:`qutip.solver.floquet.fmmesolve`  always expects the ``e_ops`` to be specified in the laboratory basis (as for other solvers) and we can calculate expectation values using:
```
